File: medicalcheckrest/WMedicalCheck/WMedicalCheck/medicalcheck/src/main/java/ch/hearc/medicalcheck/service/MedicineService.java
package ch.hearc.medicalcheck.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import ch.hearc.medicalcheck.model.Medicine;
import ch.hearc.medicalcheck.repository.MedicineRepository;

/*
* Project   : Medical Check Rest
* Authors   : William Bikuta, Milán Cerviño, Ilyas Boillat, David Oktay
* Date      : 28.01.2022
* Class     : INF3dlm-a
* */

/**
 * bridge between repository and controller
 * each methods is explained in the associate controller
 * check them for more information
 */

@Service
public class MedicineService {
	@Autowired
	private MedicineRepository medicineRepository;

	public List<Medicine> getAll() {
		return medicineRepository.findAll();
	}

	public Medicine get(Integer id) {
		return medicineRepository.findById(id).get();
	}

	public List<Medicine> getAllByIdUser(Integer iduser) {
		return medicineRepository.getAllMedicineByIdUser(iduser);
	}

	public Medicine create(Medicine medicine) {
		return medicineRepository.save(medicine);
	}

	public Medicine update(Integer idMedicine, Medicine medicine) {
		Optional<Medicine> medicineData = medicineRepository.findById(medicine.getId());

		if (medicineData.isPresent()) {
			medicine.setId(idMedicine);
			return medicineRepository.save(medicine);
		}
		return null; 
	}

	public void delete(Integer id) {
		medicineRepository.deleteById(id);
	}
}

File: medicalcheck/app/src/main/java/ch/hearc/medicalcheck/services/MedicineService.kt
package ch.hearc.medicalcheck.services

import ch.hearc.medicalcheck.models.Medicine
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

/*
* Project   : Medical Check Mobile
* Authors   : William Bikuta, Milán Cerviño, Ilyas Boillat, David Oktay
* Date      : 28.01.2022
* Class     : INF3dlm-a
* */

interface  MedicineService {
    @GET("medicines")
    fun getAll(): Call<List<Medicine>>

    @GET("medicines/{id}")
    fun get(@Path("id") id : Int): Call<Medicine>

    @GET("medicines/user/{id}")
    fun getAllByIdUser(@Path("id") id : Int): Call<List<Medicine>>

    @POST("medicines")
    fun create(@Body medicine: Medicine): Call<Medicine>

    @DELETE("medicines/{id}")
    fun delete(@Path("id") id: Int): Call<ResponseBody>

    @PUT("medicines/{id}")
    fun update(@Path("id") id : Int, @Body medicine: Medicine): Call<Medicine>
}